public/app/features/explore/Logs/LogsContainer.tsx

```diff
@@ -55,7 +57,46 @@ interface LogsContainerProps extends PropsFromRedux {
   isFilterLabelActive: (key: string, value: string) => Promise<boolean>;
 }
 
-class LogsContainer extends PureComponent<LogsContainerProps> {
+interface LogsContainerState {
+  logDetailsFilterAvailable: boolean;
+}
+
+class LogsContainer extends PureComponent<LogsContainerProps, LogsContainerState> {
+  state: LogsContainerState = {
+    logDetailsFilterAvailable: false,
+  };
+
+  componentDidMount() {
+    this.checkFiltersAvailability();
+  }
+
+  componentDidUpdate(prevProps: LogsContainerProps) {
+    this.checkFiltersAvailability();
+  }
+
+  private checkFiltersAvailability() {
+    const { queries, datasourceInstance } = this.props;
+
+    if (datasourceInstance?.modifyQuery || hasToggleableQueryFiltersSupport(queries)) {
+      this.setState({ logDetailsFilterAvailable: true });
+      return;
+    }
+
+    const promises = [];
+    for (const query of queries) {
+      if (query.datasource) {
+        promises.push(getDataSourceSrv().get(query.datasource));
+      }
+    }
+
+    Promise.all(promises).then((dataSources) => {
+      const logDetailsFilterAvailable = dataSources.some(
+        (ds) => ds.modifyQuery || hasToggleableQueryFiltersSupport(ds)
+      );
+      this.setState({ logDetailsFilterAvailable });
+    });
+  }
```

Comment:
I guess you are doing this mainly for mixed datasources, right? Not sure if we want to tackle the problem already now. In my opinion we can just add another `hasModifyQuerySupport` method and use that instead of change the state.

Mixed datasource come with a couple of other issues, e.g. Log context and I think we can think about that with a distinct task/issue. WDYT?

Reply:
I don't mind that approach, but if we can support both here, why wouldn't we? Otherwise, if you use mixed with a single logs source you will not see the 🔎  filters.

Reply:
> Otherwise, if you use mixed with a single logs source you will not see the 🔎 filters.

Yea, that's fine, right? Because with this PR we are not passing the right `datasourceInstance` with mixed datasources anyways.

https://play.grafana.org/goto/cAIKPkgIR?orgId=1 -> filter buttons are not functional.

Reply:
Not sure what's going on with Play, with I'm sure those are supposed to work:

With `toggleLabelsInLogsUI = true`
![2023-08-16 12 27 40](https://github.com/grafana/grafana/assets/1069378/0a95b2c5-7f05-44f6-8762-0da5805b545e)

With `toggleLabelsInLogsUI = false`
![2023-08-16 12 29 16](https://github.com/grafana/grafana/assets/1069378/ad273417-22b8-4219-ae54-2ac4ce428acb)

Reply:
Also works in Play:

![2023-08-16 12 31 06](https://github.com/grafana/grafana/assets/1069378/c9a62396-9313-479f-a0c8-1321ccb4fd77)

I was sure about this because I developed the changes on the 🔍  icons using Mixed with Elastic and Loki at the same time, that's why I thought of giving support by default.

Reply:
I think this is a good litmus test to show how confusing it is for users when they click on the 🔍  icons and nothing happens (filter is already present).

Reply:
Oh, lol. I did not notice I always clicked the same label. Good catch - let's disregard this comment and solve for both. We might rework this loop when we come to further mixed DS work though, but that's fine I guess.

Reply:
Sounds good! Thanks.